include/zephyr/bluetooth/services/ots.h

```diff
@@ -725,6 +725,20 @@ struct bt_ots_cb {
 	 */
 	void (*obj_name_written)(struct bt_ots *ots, struct bt_conn *conn,
 				 uint64_t id, const char *cur_name, const char *new_name);
+
+	/** @brief Object Calculate checksum callback
+	 *
+	 *  This callback is called when the OACP Calculate Checksum procedure is performed.
+	 *  Because object data is opaque to OTS service, the OTS user is the only one who
+	 *  knows where data is, user should return pointer of actual object data.
+	 *
+	 *  @param id         Object ID.
+	 *  @param offset     The first octet of the object contents need to be calculated.
+	 *  @param length     The length number of octets object name.
+	 *  @return Pointer of Object data.
+	 */
+	const uint8_t *(*obj_cal_checksum)(uint64_t id, off_t offset, size_t len);
```

Comment:
`obj_read` and `obj_write` use `void *` for the data pointers, suggest to do the same here. 

We could also consider changing the return value to `ssize_t` or `int` so that the application can return an error if something isn't correct, or it wants to reject the request for whatever reason (e.g. if the `len` is too large). 

In other words, I suggest that this works the same as the `obj_read`, including the `ots` and `conn` pointers too.

Reply:
Modify OTS client API to int bt_ots_client_get_object_checksum(struct bt_ots_client *otc_inst, struct bt_conn *conn,
				      off_t offset, size_t len);